src/components/dashboard/FaceRecognition/FaceRecognition.js

```diff
@@ -62,85 +58,51 @@ class FaceRecognition extends React.Component<FaceRecognitionProps, State> {
 
   height = 0
 
-  componentWillUnmount = () => {
-    log.debug('Unloading ZoomSDK', this.loadedZoom)
-    this.timeout && clearTimeout(this.timeout)
-  }
-
-  componentWillMount = async () => {
-    await zoomSdkLoader.ready
-    this.loadedZoom = ZoomSDK
-
-    navigator.getMedia =
-      navigator.getUserMedia || // use the proper vendor prefix
-      navigator.webkitGetUserMedia ||
-      navigator.mozGetUserMedia ||
-      navigator.msGetUserMedia
-
-    navigator.getMedia(
-      { video: true },
-      () =>
-        (this.timeout = setTimeout(() => {
-          this.setState({ zoomReady: true })
-        }, 0)),
-      this.showFRError
+  uuidv4() {
+    return ([1e7] + -1e3 + -4e3 + -8e3 + -1e11).replace(/[018]/g, c =>
+      (c ^ (crypto.getRandomValues(new Uint8Array(1))[0] & (15 >> (c / 4)))).toString(16)
     )
   }
 
-  componentDidMount = () => {}
-
-  /**
-   *  unused
-   */
-  setWidth = () => {
-    const containerWidth =
-      (this.containerRef && this.containerRef.current && this.containerRef.current.offsetWidth) || 300
-    this.width = Math.min(this.width, containerWidth)
-    this.height = window.innerHeight > window.innerWidth ? this.width * 1.77777778 : this.width * 0.5625
-    log.debug({ containerWidth, width: this.width, height: this.height })
-  }
-
-  onCaptureResult = (captureResult?: ZoomCaptureResult): void => {
-    //TODO: rami check uninitilized, return
-    log.debug('zoom capture completed', { captureResult })
+  onCaptureResult = (face, camera, images): void => {
+    log.debug('capture completed', { face, imageCount: images.length })
     fireEvent('FR_Capture')
-    if (captureResult === undefined) {
+    if (images === undefined || images.length === 0) {
       log.error('empty capture result')
       this.showFRError('empty capture result')
     } else {
-      this.startFRProcessOnServer(captureResult)
+      this.startFRProcessOnServer(images)
     }
   }
 
-  startFRProcessOnServer = async (captureResult: ZoomCaptureResult) => {
+  startFRProcessOnServer = async images => {
```

Comment:
you don't call destructured instance methods, you use this. everywhere
so you don't need to declare those methods as instance variables are arrow functions
just declare them as ES6 methods

Reply:
done